src/engraving/libmscore/edit.cpp

```diff
@@ -5224,7 +5224,9 @@ void Score::undoAddElement(EngravingItem* element, bool addToLinkedStaves, bool
             for (Score* s : scoreList()) {
                 staffList.push_back(s->staff(0)); // system objects always appear on the top staff
                 for (Staff* staff : s->getSystemObjectStaves()) {
-                    staffList.push_back(staff);
+                    if (staff->idx() != mu::nidx) {
+                        staffList.push_back(staff);
+                    }
```

Comment:
So this one is to ensure that if _still_ some non-existing staff makes it into `systemObjectStaves` (for example because it was left in there after removing a staff?)...

Reply:
That's correct--even when the staff is removed it remains in memory as part of the undo stack, I believe.

Reply:
Since @mike-spa's work about system objects, is the situation that `systemObjectStaves` contains an invalid staff still to be expected? Or should we turn this into an assertion? For example, 
```suggestion
                    IF_ASSERT_FAILED(staff->idx() != mu::nidx) {
                        continue;
                    }
                    
                    staffList.push_back(staff);
```

Reply:
Real good point! I tested that change with a rebase and it does appear that system object staves are being deleted from the list when they're removed. I'll make this change here and also the other place you indicated below.